### PR TITLE
feat: set imagePullPolicy for tigera-operator by helm install

### PR DESCRIPTION
**Pull request by lvyanru8200**
tigera-operator, calicoctl image pulling policy can be set when using helm

Fixes: https://github.com/projectcalico/calico/issues/7107
Signed-off-by: lvyanru <yanru.lv@daocloud.io>

## Todos

- [x] Tests
- [x] Documentation
- [x] Release note

## Release Note

```release-note
Support for imagePullPolicy in tigera operator helm chart.
```

## Reminder for the reviewer

Make sure that this PR has the correct labels and milestone set.

Every PR needs one `docs-*` label.

- `docs-pr-required`: This change requires a change to the documentation that has not been completed yet.
- `docs-completed`: This change has all necessary documentation completed.
- `docs-not-required`: This change has no user-facing impact and requires no docs.

Every PR needs one `release-note-*` label.

- `release-note-required`: This PR has user-facing changes. Most PRs should have this label.
- `release-note-not-required`: This PR has no user-facing changes.

Other optional labels:

- `cherry-pick-candidate`: This PR should be cherry-picked to an earlier release. For bug fixes only.
- `needs-operator-pr`: This PR is related to install and requires a corresponding change to the operator.

### CODE REVIEW

**Comment by lvyanru8200** [comment]
cc @caseydavenport

File: charts/tigera-operator/README.md [hunk 1]
```diff
@@ -146,4 +146,7 @@ tigeraOperator:
   registry: quay.io
 calicoctl:
   image: docker.io/calico/ctl
+
+# Some defaults used in the templates.
```
Comment (caseydavenport) [inline charts/tigera-operator/README.md:150]: I think this comment should explain how the imagePullPolicy is used. e.g.,

> Configures the image pull policy for all containers in the tigera/operator deployment.

Ideally, this is also used to configure the image pull policy of containers deployed by the operator but it doesn't expose that capability today IIRC.

Comment (lvyanru8200) [reply charts/tigera-operator/README.md]: Modified

**Review by caseydavenport — APPROVED** [review]

**Comment by mgleung** [comment]
@lvyanru8200 could I get you to run `make generate` and commit those changes?

**Comment by lvyanru8200** [comment]
> 

I executed make generate,Not much has changed

**Comment by lvyanru8200** [comment]
cc @mgleung

**Comment by mgleung** [comment]
@lvyanru8200 it looks like there's a 2 line change in the operator manifest that needs to be picked up before we can merge this. Sorry to ask this of you again but can you try `make generate` again?

**Comment by mgleung** [comment]
@lvyanru8200 wanted to bump this again to hopefully get that last `make generate` call to merge this in.

**Comment by mgleung** [comment]
@lvyanru8200 bumping this again to hopefully get a last `make generate` run

**Comment by caseydavenport** [comment]
Looks like we've still got conflicts on this one.

**Pull request closed without merging**